tests/integration/ha_tests/test_ha.py

```diff
@@ -65,6 +84,87 @@ async def test_build_and_deploy(ops_test: OpsTest) -> None:
+    MongoDB replica set configuration.
+    """
+    await insert_focal_to_cluster(ops_test)
+
+    # add units and wait for idle
+    app = await get_application_name(ops_test, APP_NAME)
+    await scale_application(ops_test, app, len(ops_test.model.applications[app].units) + 2)
+
+    # grab unit hosts
+    hostnames = await get_units_hostnames(ops_test)
+
+    # connect to replica set uri and get replica set members
+    member_hosts = await fetch_replica_set_members(ops_test)
+
+    # verify that the replica set members have the correct units
+    assert set(member_hosts) == set(hostnames), "all members not running under the same replset"
+
+    # verify that the no writes were skipped
+    await find_focal_in_cluster(ops_test)
+    await verify_writes(ops_test)
+
+
+@pytest.mark.abort_on_fail
+async def test_scale_down_capablities(ops_test: OpsTest, continuous_writes) -> None:
+    """Tests clusters behavior when scaling down a minority and removing a primary replica."""
+    await insert_focal_to_cluster(ops_test)
+
+    app = await get_application_name(ops_test, APP_NAME)
+    minority_count = int(len(ops_test.model.applications[app].units) / 2)
+    expected_units = len(ops_test.model.applications[app].units) - minority_count
+
+    # find leader unit
+    leader_unit = await find_unit(ops_test, leader=True)
+
+    # verify that we have a leader
+    assert leader_unit is not None, "No unit is leader"
+
+    # Force delete the leader and scale down
+    await kubectl_delete(ops_test, leader_unit, False)
+    await scale_application(ops_test, app, expected_units)
+
+    # grab unit hosts
+    hostnames = await get_units_hostnames(ops_test)
+
+    # check that the replica set with the remaining units has a primary
+    primary = await get_replica_set_primary(ops_test)
+
+    # verify that the primary is not None
+    assert primary is not None, "replica set has no primary"
+
+    # check that the primary is one of the remaining units
+    assert (
+        f"{primary.replace('/', '-')}.mongodb-k8s-endpoints" in hostnames
+    ), "replica set primary is not one of the available units"
+
+    # verify that the configuration of mongodb no longer has the deleted ip
+    member_hosts = await fetch_replica_set_members(ops_test)
+
+    # verify that the replica set members have the correct units
+    assert set(member_hosts) == set(hostnames), "mongod config contains deleted units"
+
+    # verify that the no writes were skipped
+    await find_focal_in_cluster(ops_test)
+    await verify_writes(ops_test)
+
+
+async def test_replication_across_members(ops_test: OpsTest, continuous_writes) -> None:
+    """Check consistency, ie write to primary, read data from secondaries."""
+    # first find primary, write to primary, then read from each unit
+    await insert_focal_to_cluster(ops_test)
+
+    # verify that the no writes were skipped
+    await find_focal_in_cluster(ops_test)
+    await verify_writes(ops_test)
```

Comment:
AFAICT `verify_writes` implicitly checks that each replica has the same data. Hence there is no need to also check that focal is replicated across the cluster
```suggestion
    # verify that no writes were missed to the DB and that writes are consistent across all members
    await verify_writes(ops_test)
```

Reply:
They check different DBs. `verify_writes()` checks `db["continuous_writes_database"]["test_collection"]`, while `find_focal_in_cluster()` checks `db["new-db"]["test_ubuntu_collection"]`.

Reply:
Removing `insert_focal_to_cluster()` and `find_focal_in_cluster()`